packages/datepickers/src/DatepickerRange/components/Month.tsx

```diff
@@ -0,0 +1,303 @@
+    }
+
+    let isHighlighted = false;
+
+    if (startValue !== undefined && endValue !== undefined) {
+      isHighlighted =
+        (isAfter(date, startValue) || isSameDay(date, startValue)) &&
+        (isBefore(date, endValue) || isSameDay(date, endValue)) &&
+        !isSameDay(startValue, endValue);
+    } else if (startValue !== undefined && hoverDate !== undefined) {
+      isHighlighted =
+        (isAfter(date, startValue) || isSameDay(date, startValue)) &&
+        (isBefore(date, hoverDate) || isSameDay(date, hoverDate));
+    }
+
+    const isHighlightStart = (isHighlighted && startValue && isSameDay(date, startValue)) || false;
+
+    const isHighlightEnd =
+      (isHighlighted && (endValue && isSameDay(date, endValue))) ||
+      (hoverDate && isSameDay(date, hoverDate) && !isBefore(date, endValue!)) ||
+      false;
+
+    let isInvalidDateRange =
+      (endValue &&
+        startValue &&
+        (compareAsc(endValue, startValue) === -1 || compareAsc(endValue, startValue) === 0)) ||
+      false;
+
+    if (minValue) {
+      if (startValue) {
+        isInvalidDateRange =
+          isInvalidDateRange || compareAsc(startValue, subDays(minValue, 1)) === -1;
+      }
+
+      if (endValue) {
+        isInvalidDateRange =
+          isInvalidDateRange || compareAsc(endValue, subDays(minValue, 1)) === -1;
+      }
+    }
+
+    if (maxValue) {
+      if (startValue) {
+        isInvalidDateRange = isInvalidDateRange || compareAsc(startValue, maxValue) === 1;
+      }
+
+      if (endValue) {
+        isInvalidDateRange = isInvalidDateRange || compareAsc(endValue, maxValue) === 1;
+      }
+    }
+
+    return (
+      <StyledCalendarItem key={`day-${itemsIndex}`} isSmall={small}>
+        <StyledHighlight
+          isHighlighted={!isInvalidDateRange && isHighlighted}
+          isStart={!isInvalidDateRange && isHighlightStart}
+          isEnd={!isInvalidDateRange && isHighlightEnd}
+          data-test-id="highlight"
+          data-test-highlighted={!isInvalidDateRange && isHighlighted}
+          data-test-start={!isInvalidDateRange && isHighlightStart}
+          data-test-end={!isInvalidDateRange && isHighlightEnd}
+        />
+        <StyledDay
+          isToday={isCurrentDate}
+          isPreviousMonth={isPreviousMonth}
+          isSelected={!isInvalidDateRange && isSelected}
+          isDisabled={isDisabled}
+          small={small!}
+          onClick={() => {
+            if (!isDisabled) {
+              dispatch({ type: 'CLICK_DATE', value: date });
+            }
+          }}
+          onMouseEnter={() => {
+            if (!isSelected) {
+              dispatch({ type: 'HOVER_DATE', value: date });
+            }
+          }}
+          data-test-id="day"
+          data-test-previous={isPreviousMonth}
+          data-test-selected={!isInvalidDateRange && isSelected}
+          data-test-disabled={isDisabled}
+          data-test-today={isCurrentDate}
+          data-test-hidden="false"
```

Comment:
What's the purpose of all of these for testing only?

Reply:
With us no longer using `enzyme` we aren't able to assert against props. This allows us to ensure the states are being applied without having to assert against a specific styled-components style.  This is the same process as the `Datepicker` PR.